content/snapshot-resource/index.md

```diff
@@ -0,0 +1,255 @@
+---
+rfc: 0025
+start_date: 2018-10-03
+decision_date:
+pr: openregister/registers-rfcs#42
+status: draft
+---
+
+# Snapshot resource
+
+## Summary
+
+This RFC proposes creating a new Snapshot resource and redefine the Record
+resource to have a single responsibility.
+
+This is a breaking change.
+
+## Motivation
+
+One of the most difficult concepts to explain in the Registers ecosystem is
+the concept of “record”. The reasons boil down to two things: First, the
+concept of “record” has shifted over time as a natural adaptation to how
+Registers has evolved and second the current record definition doesn't match
+the expectations on the record resource. Probably the second is consequence of
+the first.
+
+Let me elaborate on what does “the current record definition doesn't match
+the expectations on the record resource” mean.
+
+The current definition of record is: The most recent entry for a given key.
+The record resource though is the combination of the most recent entry for a
+given key and the blob of data referenced by it.
+
+The reason for having the blob of data merged in is to make the record
+resource useful to the type of user that wants to get the data for a given
+key.
+
+The reason for having the entry data is to provide the context for how it
+relates to the log of entries. Useful to users that need to deal with
+entries, blobs and the log itself.
+
+The first reason is “porcelain” in the sense of “don't make me think, give me
+the data”. The second is “plumbing” in the sense of “let me handle the low
+level bits myself”.
+
+The result is tension between these two needs and an ill-defined concept.
+The proposal is to split both needs into dedicated resources.
+
+## Explanation
+
+Let's define “snapshot” to be the set of most recent entries for each unique key.
```

Comment:
I think it would be clearer to say:
*snapshot as a set containing the most recent entry for each key.*